### PR TITLE
Fix bugs in config.xml handling caused by xml-mapping component

### PR DESCRIPTION
**Pull request by jsoref**
This fixes issue #223, #231, and #235

### CODE REVIEW

**Comment by jsoref** [comment]
We got confirmation from the reporter of #235 that this does indeed fix the bug.

File: routes/project_config.js [hunk 1]
```diff
@@ -28,18 +28,23 @@ module.exports = {
...
-                            "/widget/icon",
-                            "/widget/rim:splash",
-                            "/widget/rim:permissions/rim:permit"
-                        ]
-                    })
-                });
+                try {
+                    res.send(200, {
+                        success: !error,
+                        configFile: xm.load((data || "").replace(/^\uFEFF/, ''), {
```
Comment (rcruz) [inline routes/project_config.js:34]: Is that a special char?

Comment (rcruz) [inline routes/project_config.js:34]: Ah, that is the unicorn. Are we aware of more types of unicorns in the world?

Comment (jsoref) [inline routes/project_config.js:34]: That's the only one I know of.

The other code here will at least result in us getting an error at the UI side. -- Although the bits won't be escaped, so most unicorns will be invisible in the error message (but they'll be correctly line/column identified, so someone could eventually find them...).

**Comment by swypych** [comment]
r+

**Pull request merged**